src/source_util.cpp

```diff
@@ -121,21 +84,30 @@ SourceUtil::expandRange(const clang::SourceRange &range,
   // want to resolve to the entire macro call.
 
   clang::SourceLocation begin = range.getBegin();
+  while (sourceManager.isMacroArgExpansion(begin)) {
+    begin = sourceManager.getImmediateSpellingLoc(begin);
+  }
+
   if (sourceManager.isMacroBodyExpansion(begin)) {
     begin = sourceManager.getExpansionRange(begin).getBegin();
   } else {
-    begin = sourceManager.getFileLoc(begin);
+    begin = sourceManager.getSpellingLoc(begin);
   }
 
   // Get the end location, resolving from macro definition to macro call
   // location. The end location of a statement points to the beginning of the
   // last token in it, so we must use the lexer to traverse the token too.
 
   clang::SourceLocation end = range.getEnd();
+
+  while (sourceManager.isMacroArgExpansion(end)) {
+    end = sourceManager.getImmediateSpellingLoc(end);
+  }
+
   if (sourceManager.isMacroBodyExpansion(end)) {
     end = sourceManager.getExpansionRange(end).getEnd();
   } else {
-    end = sourceManager.getFileLoc(end);
```

Comment:
The issue was here, `getFileLoc` always returns the first token in an expansion, while we want the last one here. By decomposing this into `getImmediateSpellingLoc` and `getExpansionRange().getEnd()` we are able to get the correct location instead.